product_supplier_code_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-17 Eficent Business and IT Consulting Services, S.L.
+#           (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    product_supplier_code = fields.Char(
+        string='Product Supplier Code',
+        compute='_compute_product_supplier_code'
```

Comment:
Should be great to have a function inv defined, to have the possibility, from the purchase order, to change the supplier code defined in the supplierinfo.

don't you think ? 

regards.